### PR TITLE
search: inline SearchFilesInRepos

### PR DESCRIPTION
**Pull request by rvantonder**
## Test plan
Semantics-preserving.

### CODE REVIEW

File: internal/search/textsearch/textsearch.go [hunk 1]
```diff
@@ -103,7 +90,21 @@ func (t *RepoSubsetTextSearch) Run(ctx context.Context, db database.DB, stream s
...
+				return request.Search(ctx, stream)
+			})
+		}
+
+		// Concurrently run searcher for all unindexed repos regardless whether text or regexp.
+		g.Go(func() error {
+			return searcher.SearchOverRepos(ctx, t.SearcherArgs, stream, request.UnindexedRepos(), false)
+		})
+
+		return g.Wait()
```
Comment (rvantonder) [inline internal/search/textsearch/textsearch.go:107]: Having this inside the `RepoSubsetTextSearch` job helps _massively_, and the main reason for this change. The two goroutines here are effectively the `searcher` <-> `zoekt` jobs I want, and the only thing remaining is partitioning indexed/unindexed repos before this

File: internal/search/textsearch/textsearch.go [hunk 2]
```diff
@@ -19,39 +19,26 @@ import (
...
 var MockSearchFilesInRepos func() ([]result.Match, *streaming.Stats, error)
 
-// SearchFilesInRepos searches a set of repos for a pattern.
-func SearchFilesInRepos(ctx context.Context, zoektArgs zoektutil.IndexedSearchRequest, searcherArgs *search.SearcherParameters, notSearcherOnly bool, stream streaming.Sender) (err error) {
-	if MockSearchFilesInRepos != nil {
-		matches, mockStats, err := MockSearchFilesInRepos()
-		stream.Send(streaming.SearchEvent{
-			Results: matches,
-			Stats:   mockStats.Deref(),
-		})
```
Comment (rvantonder) [inline internal/search/textsearch/textsearch.go:29]: Interestingly enough, no changes are needed to tests and they still pass. The way this logic was triggered via tests before was only via `SearchFilesInReposBatch`, and it only checks on err, so 🤷 . With the new structure things will be more testable and I'm happy to get rid of jank that doesn't seem to do anything anyway.

File: internal/search/textsearch/textsearch.go [hunk 3]
```diff
@@ -19,39 +19,26 @@ import (
...
-		return err
-	}
+// SearchFilesInReposBatch is a convenience function around searchFilesInRepos
+// which collects the results from the stream.
+func SearchFilesInReposBatch(ctx context.Context, request zoektutil.IndexedSearchRequest, searcherArgs *search.SearcherParameters, searcherOnly bool) ([]*result.FileMatch, streaming.Stats, error) {
```
Comment (rvantonder) [inline internal/search/textsearch/textsearch.go:24]: This is only used by the `repoHasFile` logic. I will inline that next too.

**Review by camdencheek — APPROVED** [review]

**Pull request merged**